tools/roslaunch/src/roslaunch/__init__.py

```diff
@@ -295,6 +303,9 @@ def main(argv=sys.argv):
                 p = roslaunch_parent.ROSLaunchParent(uuid, args, roslaunch_strs=roslaunch_strs,
                         is_core=options.core, port=options.port, local_only=options.local_only,
                         verbose=options.verbose, force_screen=options.force_screen)
+                if options.core:
+                    p.num_workers = options.num_workers
+                    p.timeout = options.timeout
```

Comment:
Please don't add additional members to the `ROSLaunchParent` instance but add them to the class.